pkg/destroy/gcp/dns.go

```diff
@@ -105,17 +114,19 @@ func possibleZoneParents(dnsDomain string) []string {
 	return result
 }
 
-func getParentDNSZone(dnsDomain string, publicZones []dnsZone, logger logrus.FieldLogger) *dnsZone {
+func getParentDNSZones(dnsDomain string, publicZones []dnsZone, logger logrus.FieldLogger) []*dnsZone {
+	possibleParentZones := []*dnsZone{}
```

Comment:
Nit: This name is very close to `possibleZoneParents`. Since you're checking in the loop below, can this just be `parentZones`?